pkg/privilege/privileges/cache.go

```diff
@@ -260,6 +262,26 @@ func (g roleGraphEdgesTable) Find(user, host string) bool {
 	return ok
 }
 
+type immutable struct {
+	user         []UserRecord
+	db           []dbRecord
+	tablesPriv   []tablesPrivRecord
+	columnsPriv  []columnsPrivRecord
+	defaultRoles []defaultRoleRecord
+
+	globalPriv  []globalPrivRecord
```

Comment:
Why need to keep a immutable `globalPriv` and a mutable `Global`? Any problem to just keep one `Global` field?

Reply:
Make the apply diff logic simpler if we're using one single slice representation... 
we can copy and paste a lot code and just handle them the same way.

It's possible to keep different representation for different field, or even using Map for almost all field, but we need to add more code.